lib/rex/random_identifier/generator.rb

```diff
@@ -34,10 +34,40 @@ class ExhaustedSpaceError < StandardError; end
     # This should be pretty universal for identifier rules
     :char_set => Rex::Text::AlphaNumeric+"_",
     :first_char_set => Rex::Text::LowerAlpha,
-    :forbidden => []
+    :forbidden => [].freeze
+  }
+
+  JavaOpts = DefaultOpts.merge(
+    forbidden: (
+      DefaultOpts[:forbidden] +
+      %w[
+        abstract assert boolean break byte case catch char class const
+        continue default do double else enum extends false final finally
+        float for goto if implements import instanceof int interface long
+        native new null package private protected public return short
+        static strictfp super switch synchronized this throw throws
+        transient true try void volatile while
```

Comment:
```suggestion
        transient true try void volatile while _
```

Java file:
```
class Main {  
  public static void main(String args[]) { 
	String _ = "hello world";
       System.out.println(_); 
  } 
}
```

Example error:
```
Main.java:4: error: as of release 9, '_' is a keyword, and may not be used as an identifier
    System.out.println(_); 
                       ^
```